react/features/base/tracks/middleware.js

```diff
@@ -67,6 +72,46 @@ MiddlewareRegistry.register(store => next => action => {
 
         break;
     }
+    case SET_VISIBLE_REMOTE_PARTICIPANTS: {
+        const state = store.getState();
+
+        if (!getSourceNameSignalingFeatureFlag(state)) {
+            break;
+        }
+
+        const { visibleRemoteParticipants, remoteParticipants } = state['features/filmstrip'];
```

Comment:
This still assume there is only one video stream per participant. Once we support multi stream we might need to decouple the `visibleRemoteParticipants` logic with a `visibleRemoteTracks` equivalent

Reply:
Yes, I think that's fine to assume for now, because the plan is to create a separate tile for the screen stream. The change can be part of the PR which adds the extra thumbnail.

Reply:
Agreed, as I started looking into the code I realized how tightly coupled tiles are to participants

Reply:
I was actually hoping that the UI components which are using the track streaming status would use a hook which registers the listener locally. Can you try looking if that would be possible? It would be much simpler logic.

Reply:
I think it would require `_renderIcon` to be changed into a functional component. What are your thoughts on this?

Reply:
Ahh yes I'll give that a try, that will greatly simplify the logic. I remembering seeing `isPlayable` and `isVideoRenderable` being determined based on `JitsiParticipantConnectionStatus.ACTIVE`. I'll double check to make sure there aren't any race conditions. 

On another note, are there any recommendations on what to use for testing the jitsi-meet frontend code?

Reply:
Do you mean how to test different "track streaming status" states? In order to do that you will probably need at least 2 different machines where you can disconnect one from the network while the other one is online. Network link conditioner tool will be useful to trigger endpoints falling out of "forwarded sources".

Reply:
For general tests probably torture if there are any tests impact by those changes https://github.com/jitsi/jitsi-meet-torture